Holovibes/sources/compute/fourier_transform.cc

```diff
@@ -274,93 +274,39 @@ void FourierTransform::insert_moments()
-                    cudaXMemcpy(moments_env_.moment0_buffer + offset0, src_buf, image_size, cudaMemcpyDeviceToDevice);
-                    offset0 += image_resolution;
-                }
-                else if (i % 3 == 1)
-                {
-                    // Image goes to moment 1
-                    cudaXMemcpy(moments_env_.moment1_buffer + offset1, src_buf, image_size, cudaMemcpyDeviceToDevice);
-                    offset1 += image_resolution;
-                }
-                else
-                {
-                    // Image goes to moment 2
-                    cudaXMemcpy(moments_env_.moment2_buffer + offset2, src_buf, image_size, cudaMemcpyDeviceToDevice);
-                    offset2 += image_resolution;
-                }
-                src_buf += image_resolution;
-            }
         });
 }
 
 void FourierTransform::insert_moments_to_output()
 {
-    fn_compute_vect_.push_back(
+    fn_compute_vect_->conditional_push_back(
         [this]()
         {
             size_t image_resolution = fd_.get_frame_res();
             size_t image_size = image_resolution * sizeof(float);
             if (setting<settings::ImageType>() == ImgType::Moments_0)
             {
-                cudaXMemcpy(buffers_.gpu_postprocess_frame.get(),
-                            moments_env_.moment0_buffer.get(),
-                            image_size,
-                            cudaMemcpyDeviceToDevice);
+                cudaXMemcpyAsync(buffers_.gpu_postprocess_frame.get(),
+                                 moments_env_.moment0_buffer.get(),
+                                 image_size,
+                                 cudaMemcpyDeviceToDevice,
+                                 stream_);
             }
             else if (setting<settings::ImageType>() == ImgType::Moments_1)
             {
-                cudaXMemcpy(buffers_.gpu_postprocess_frame.get(),
-                            moments_env_.moment1_buffer.get(),
-                            image_size,
-                            cudaMemcpyDeviceToDevice);
+                cudaXMemcpyAsync(buffers_.gpu_postprocess_frame.get(),
+                                 moments_env_.moment1_buffer.get(),
+                                 image_size,
+                                 cudaMemcpyDeviceToDevice,
+                                 stream_);
             }
             else if (setting<settings::ImageType>() == ImgType::Moments_2)
             {
-                cudaXMemcpy(buffers_.gpu_postprocess_frame.get(),
-                            moments_env_.moment2_buffer.get(),
-                            image_size,
-                            cudaMemcpyDeviceToDevice);
+                cudaXMemcpyAsync(buffers_.gpu_postprocess_frame.get(),
+                                 moments_env_.moment2_buffer.get(),
+                                 image_size,
+                                 cudaMemcpyDeviceToDevice,
+                                 stream_);
```

Comment:
You can refacto this by changing only the source buffer in a variable and then call cudaMemcpy

Reply:
Done